### ISSUE TITLE
Perl5 layer unusable due to inserting extra parens / brackets

### ISSUE THREAD

**Issue by rvbdpudar** [syl20bnr/spacemacs#11815]
#### Description :octocat:

When using the perl5 layer, typing near keywords like 'while' and 'if' is unusable.

#### Reproduction guide :beetle:
- Start Emacs
- Make sure perl5 layer is installed
- Start a new Perl file
- Enter insert mode and type 'while'
- Exit insert mode. The parens and curly brackets are auto inserted
- Enter insert mode with the cursor after the 'while' but before the closing curly bracket
- Exit insert mode. A new set of parens and curly brackets are auto inserted
- Throw keyboard

*Observed behaviour:* :eyes: :broken_heart:
Step 1: type 'while'

while

Step 2: exit insert mode, parens and brackets inserted

while () {
}

Step 3: enter insert mode and exit insert mode

while () {
} () {
}

Step 4: repeat Step 3:

while () {
} () {
} () {
}

*Expected behaviour:* :heart: :smile:

Step 3 should do nothing

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 2c28c7d21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion better-defaults emacs-lisp git markdown multiple-cursors neotree org shell spell-checking syntax-checking perl5)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by collector1871** [comment]
i fight with this error some time ago and found a solution.

cperl mode uses abbrevation which is not really compatible with yassnippets or/and autocomplete 
so you need to turn off abbreviation and clear abbreviation table

you can check all abbrevs via `M-x edit-abbrevs` : https://www.gnu.org/software/emacs/manual/html_node/emacs/Editing-Abbrevs.html

and you will see that cperl is using many abbrevs but also `electric functions` which are pinned to all abbrevs

basically it;s not an issue, but "tradition" - cperl mode is using abbrevs.
in my case i think that it;s better to use yasnippets/autocomplete layer:
https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Bcompletion/auto-completion

**Comment by rvbdpudar** [comment]
Thanks @collector1871, I'll give this a shot and see if it works. Maybe I'll make a PR to turn these off automatically in the `perl5` layer.
